packages/web/lib/hooks.ts

```diff
@@ -0,0 +1,39 @@
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+import { useRouter } from 'next/router';
+import { useContext, useEffect } from 'react';
+
+import { Web3Context } from '../contexts/Web3Context';
+import { useGetMeQuery } from '../graphql/autogen/types';
+
+export const useWeb3 = () => useContext(Web3Context);
+
+type UseUserOpts = {
+  redirectTo?: string;
+  redirectIfFound?: boolean;
+};
+
+export const useUser = ({ redirectTo, redirectIfFound }: UseUserOpts = {}) => {
```

Comment:
```suggestion
export type User = Pick<Me, 'id' | 'ethereum_address' | 'username'> & {
   player?: Maybe<PlayerFragmentFragment>;
};

export const useUser = ({ redirectTo, redirectIfFound }: UseUserOpts = {}): User => {
```

Reply:
IMO this is just adding unnecesarry boilerplate and duplicating what we already have in the generated types. Makes the code harder to understand.